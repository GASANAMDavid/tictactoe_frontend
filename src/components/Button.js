import React from "react";

function Button({ color, text, onClick }) {
  return (
    <div>
      <button type='button'
        onClick={onClick}
        style={{ backgroundColor: color }}
        className='btn-create'
      >
        {text}
      </button>
    </div>
  );
}

export default Button;
