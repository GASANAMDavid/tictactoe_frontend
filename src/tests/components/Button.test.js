import React from "react";
import { shallow } from "enzyme";
import Button from "../../components/Button";

describe("Button", () => {
  let button;
  const mockCreateGame = jest.fn();
  beforeEach(() => {
    button = shallow(
      <Button color='green' text='Test' onClick={mockCreateGame} />
      
    );
  });

  it("sets button text from props", () => {
    expect(button.find(".btn").text()).toEqual("Test");
  });

  it("calls mockCreatGame function when clicked", () => {
    button.find(".btn").simulate("click");
    expect(mockCreateGame).toHaveBeenCalled();
  });
});
