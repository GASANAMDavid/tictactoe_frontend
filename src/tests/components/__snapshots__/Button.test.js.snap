// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <button
    className="btn-create"
    onClick={[MockFunction]}
    style={
      Object {
        "backgroundColor": "green",
      }
    }
    type="button"
  >
    Test
  </button>
</div>
`;
